polling/frequent/activities.go

```diff
@@ -0,0 +1,33 @@
+package frequent
+
+import (
+	"context"
+	"errors"
+	"github.com/temporalio/samples-go/polling"
+	"go.temporal.io/sdk/activity"
+	"time"
+)
+
+type PollingActivities struct {
+	TestService  *polling.TestService
+	PollInterval time.Duration
+}
+
+// DoPoll Activity.
+// In this activity polling is implemented within the activity itself and not the workflow,
+// using the heartbeat mechanism to keep the activity alive
+func (a *PollingActivities) DoPoll(ctx context.Context) (string, error) {
+	for {
+		res, err := a.TestService.GetServiceResult(ctx)
```

Comment:
Mentioned in another comment, but this is the problem with using errors for actual errors and for not-done-yet poll responses. This error is swallowed. What if a serious error is happening when getting service result? There are no logs. But you probably don't want to log the not-done-yet error, just serious errors (and maybe return them so activity retry mechanism can kick in).